compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentCppWriter.scala

```diff
@@ -352,8 +350,8 @@ case class ComponentCppWriter (
          |
          |    enum {
          |      // Max. message size = size of data + message id + port
-         |      SERIALIZATION_SIZE =
-         |        sizeof(BuffUnion) +
+         |      SERIALIZATION_SIZE =${if (getBuffUnion.nonEmpty) """
```

Comment:
I think this change is not necessary. This function is called only if the component is not passive. In that case, BuffUnion cannot be empty. See https://github.com/fprime-community/fpp/blob/26e5ace8d17807fa5caf726f70636fea4d0a6e36/compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentCppWriter.scala#L261-L278.

Reply:
Why it cannot be empty? 

For example, a component could have only ports with no arguments. --> This will cause the `getBuffUnion` to return nothing. --> `sizeOf(BuffUnion)` cannot exist because `BuffUnion` doesn't exist in the code.
Maybe I can add a UT to show the case

Reply:
OK, good point. Yes, can you add a unit test to cover the case you identified? Thanks.

Reply:
I added the test